jnigen/test/summary_test.dart

```diff
@@ -0,0 +1,324 @@
+  }
+
+  test('return types', () {
+    final example = classes.getExampleClass();
+    expect(example.getMethod("getNumber").returnType.shorthand, equals("int"));
+    expect(example.getMethod("getName").returnType.shorthand,
+        equals("java.lang.String"));
+    expect(example.getMethod("getNestedInstance").returnType.name,
+        equals("$simplePackage.Example\$Nested"));
+    final listType = example.getMethod("getList").returnType;
+    assertToBeStringListType(listType);
+  });
+
+  test('parameter types', () {
+    final example = classes.getExampleClass();
+    final joinStrings = example.getMethod('joinStrings');
+    final listType = joinStrings.params[0].type;
+    assertToBeStringListType(listType);
+    final stringType = joinStrings.params[1].type;
+    expect(stringType.kind, Kind.declared);
+    expect((stringType.type as DeclaredType).binaryName, 'java.lang.String');
+  });
+
+  test('Parameters of several types', () {
+    final example = classes.getExampleClass();
+    final m = example.getMethod('methodWithSeveralParams');
+    expect(m.typeParams, hasLength(1));
+    expect(m.typeParams[0].name, 'T');
+    expect(m.typeParams[0].bounds[0].name, 'java.lang.CharSequence');
+
+    final ch = m.params[0];
+    expect(ch.type.kind, equals(Kind.primitive));
+    expect(ch.type.name, equals('char'));
+
+    final s = m.params[1];
+    expect(s.type.kind, equals(Kind.declared));
+    expect(
+        (s.type.type as DeclaredType).binaryName, equals('java.lang.String'));
+
+    final a = m.params[2];
+    expect(a.type.kind, equals(Kind.array));
+    expect((a.type.type as ArrayType).type.name, equals('int'));
+
+    final t = m.params[3];
+    expect(t.type.kind, equals(Kind.typeVariable));
+    expect((t.type.type as TypeVar).name, equals('T'));
+
+    final lt = m.params[4];
+    expect(lt.type.kind, equals(Kind.declared));
+    final listType = (lt.type.type as DeclaredType);
+    expect(listType.binaryName, equals('java.util.List'));
+    expect(listType.params, hasLength(1));
+    final tType = listType.params[0];
+    expect(tType.kind, Kind.typeVariable);
+    expect((tType.type as TypeVar).name, equals('T'));
+
+    final wm = m.params[5];
+    expect(wm.type.kind, equals(Kind.declared));
+    final mapType = (wm.type.type as DeclaredType);
+    expect(mapType.binaryName, equals('java.util.Map'));
+    expect(mapType.params, hasLength(2));
+    final strType = mapType.params[0];
+    expect(strType.name, 'java.lang.String');
+    // TODO(#141): Wildcard implementation.
+    /*
+    final wildcardType = mapType.params[1];
+    expect(wildcardType.kind, equals(Kind.wildcard));
+    expect((wildcardType.type as Wildcard).extendsBound?.name,
+        equals('java.lang.CharSequence'));
+    */
+  });
+
+  test('superclass', () {
+    final baseClass = classes.getClass('inheritance', 'BaseClass');
+    expect(baseClass.typeParams, hasLength(1));
+    final tp1 = baseClass.typeParams[0];
```

Comment:
`final typeParam = typeParams.single` gives you the only element and throws if the length != 1. Also `tp1` is a bit cryptic.